backend/lcfs/web/api/user/session.py

```diff
@@ -67,78 +48,87 @@ async def query_users(
         # Build the base query statement
         conditions = []
         if username:
-            conditions.append(func.lower(UserProfile.username).like(f"%{username.lower()}%"))
+            conditions.append(
+                func.lower(UserProfile.username).like(f"%{username.lower()}%")
+            )
         if organization:
-            conditions.append(func.lower(UserProfile.organization.name).like(f"%{organization.lower()}%"))
+            conditions.append(
+                func.lower(UserProfile.organization.name).like(
+                    f"%{organization.lower()}%"
+                )
+            )
         # if not include_inactive:
         #     conditions.append(UserProfile.is_active.is_(True))
 
-        # Sorting direction
-        sort_function = desc if pagination["sort_direction"] == "desc" else asc
-        sort_field = pagination["sort_field"]
-        offset = pagination["page"] * pagination["per_page"]
-        limit = pagination["per_page"]
+        # Apply pagination and sorting parameters
+        offset = 0 if (pagination.page < 1) else (pagination.page - 1) * pagination.size
+        limit = pagination.size
 
         # Applying pagination, sorting, and filters to the query
         query = (
             select(UserProfile)
+            .options(joinedload(UserProfile.organization))
             .where(and_(*conditions))
-            .order_by(sort_function(getattr(UserProfile, sort_field)))
-            .offset(offset)
-            .limit(limit)
         )
 
+        total_count = await self.session.scalar(
+            select(func.count(UserProfile.user_profile_id)).where(and_(*conditions))
+        )
+        # Sort the query results
+        for order in pagination.sortOrders:
+            sort_method = asc if order.direction == "asc" else desc
+            query = query.order_by(sort_method(order.field))
+
         # Execute the query
-        user_results = await self.session.execute(query)
-        results = user_results.scalars().all()
+        user_results = await self.session.execute(query.offset(offset).limit(limit))
+        # total_count = query.count()
+        results = user_results.scalars().unique().all()
 
         # Convert the results to UserBase schemas
-        return [UserBase.model_validate(user) for user in results]
+        return [UserBase.model_validate(user) for user in results], total_count
+# TODO: Need to redine search, create and modify endpoints
```

Comment:
Does this need to be done now?

Reply:
yes, create and edit endpoints will add as we refine the UI, as this ticket's main focus was ag-grid.